src/worker/index.js

```diff
@@ -69,14 +69,20 @@ function getTrackerFromUrl(url, origin) {
   return null;
 }
 
+// Storage "Options" initialization
+
 let options = {};
 
-async function updateOptions() {
+(async function () {
   const storage = await chrome.storage.local.get(['options']);
   options = storage.options || {};
-}
+})();
 
-updateOptions();
+chrome.storage.onChanged.addListener(function (changes) {
```

Comment:
I was too optimistic :( As the `chrome.storage.onChanged` API exists on Safari, it just is not triggered when options are updated. @chrmod Can you look at here, maybe I made some obvious mistake here? 

BTW, in Chrome of course it works like a charm :P.

Reply:
https://developer.apple.com/forums/thread/670007

Reply:
@chrmod If you don't see anything else, which could help, I will revert to the `sendMessage` way.